docs/pages/versions/unversioned/sdk/device.md

```diff
@@ -0,0 +1,352 @@
+
+### `Device.supportedCpuArchitectures`
+
+A list of supported processor architecture versions. The device expects the binaries it runs to be compiled for one of these architectures. The returned value will be `null` if the supported architectures could not be determined.
+
+#### Examples
+
+```js
+Device.supportedCpuArchitectures; // [ "arm64 v8", "Intel x86-64h Haswell", "arm64-v8a", "armeabi-v7a", "armeabi" ]
+```
+
+### `Device.osBuildId`
+
+The build ID of the OS that more precisely identifies the version of the OS. On Android, this corresponds to `Build.DISPLAY` (not `Build.ID`) and currently is a string as described [here](https://source.android.com/setup/start/build-numbers). On iOS, this corresponds to `kern.osversion` and is the detailed OS version sometimes displayed next to the more human-readable version.
+
+#### Examples
+
+```js
+Device.osBuildId; // iOS: "16F203"; Android: "PSR1.180720.075"
+```
+
+### `Device.osInternalBuildId`
+
+The internal build ID of the OS running on the device. On Android, this corresponds to `Build.ID`. On iOS, this is the same value as [`Device.osBuildId`](#deviceosbuildid).
+
+#### Examples
+
+```js
+Device.osInternalBuildId; // iOS: "16F203"; Android: "MMB29K"
+```
+
+### `Device.osName`
+
+The name of the OS running on the device.
+
+#### Examples
+
+```js
+Device.osName; // iOS: "iOS" or iPadOS; Android: "Android"
+```
+
+### `Device.osVersion`
+
+The human-readable OS version string. Note that the version string may not always contain three numbers separated by dots.
+
+#### Examples
+
+```js
+Device.osVersion; // iOS: 12.3.1; Android: 4.0.3
+```
+
+### `Device.platformApiLevel`
+
+**Android only.** The SDK version of the software currently running on this hardware device. This value never changes while a device is booted, but it may increase when the hardware manufacturer provides an OS update. See [here](https://developer.android.com/reference/android/os/Build.VERSION_CODES.html) to see all possible version codes and corresponding versions.
+
+#### Examples
+
+```js
+Device.platformApiLevel; // iOS: null; Android: 19
+```
+
+### `Device.osBuildFingerprint`
+
+**Android only.** A string that uniquely identifies the build of the currently running system OS. On Android, it follows this template:
+
+`$(BRAND)/$(PRODUCT)/$(DEVICE)/$(BOARD):$(VERSION.RELEASE)/$(ID)/$(VERSION.INCREMENTAL):$(TYPE)/\$(TAGS)`
+
+#### Examples
+
+```js
+Device.osBuildFingerprint; // iOS: null; Android: "google/sdk_gphone_x86/generic_x86:9/PSR1.180720.075/5124027:user/release-keys"
+```
+
+### `Device.designName`
+
+**Android only.** The specific configuration or name of the industrial design. It represents the device's name when it was designed during manufacturing into mass production. On Android, it corresponds to [`Build.DEVICE`](https://developer.android.com/reference/android/os/Build#DEVICE).
+
+#### Examples
+
+```js
+Device.designName; // iOS: null; Android: "kminilte"
+```
+
+### `Device.productName`
```

Comment:
if we're sure it really does serve the same purpose as `modelId` (a unique identifier for the device's hardware model) then I think folding it into `modelId` makes sense. I wasn't able to find a concrete explanation of what this field means to Android, though, and unless we're really sure, I'm inclined to think that it's better to keep them separate since they will both be such platform-specific values.